### PR TITLE
Admedia usersync

### PR DESCRIPTION
**Pull request by devmusings**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

Added usersync to Admedia's bidder.

- contact email of the adapter’s maintainer
- [x] official adapter submission

## Other information

Suggested updates per #1864

### CODE REVIEW

File: modules/admediaBidAdapter.js [hunk 1]
```diff
@@ -67,6 +69,16 @@ var AdmediaAdapter = function AdmediaAdapter() {
 
       adloader.loadScript(endpoint);
     }
+
+    if (userSync.iframeEnabled) {
+      userSync.registerSync('iframe', 'admedia', '//b.admedia.com/sync/iframe/');
```
Comment (harpere) [inline modules/admediaBidAdapter.js]: `userSync.iframeEnabled` does not exist.  You'd have to look at the config directly, like `config.getConfig('userSync');`

**Comment by harpere** [comment]
unit tests should be added for the change.

File: modules/admediaBidAdapter.js (hunk 1 above)
Comment (devmusings) [reply modules/admediaBidAdapter.js]: Updated to use config.

**Comment by devmusings** [comment]
Using config object and added unit tests.

**Comment by harpere** [comment]
@devmusings  looks like there's a conflict in `admediaBidAdapter.js` that needs to be resolved.

File: test/spec/modules/admediaBidAdapter_spec.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      adapter = new Adapter();
+      adapter.callBids(request_single_adunit);
+      sandbox.stub(userSync, 'triggerUserSyncs');
+    });
+
+    afterEach(() => {
+    });
+
+    it('should call usersync after default 3000 seconds', () => {
+      window.setTimeout(function() {
```
Comment (harpere) [inline test/spec/modules/admediaBidAdapter_spec.js]: A couple of problems with the setTimeout here.  
1.  you're not using an async callback (usually called `done`) to notify the framework the test is finished.
2. if this works, its adding a 3 second delay every time the tests are run.

I think a better approach would be to test that your `usersync.registerSync()` calls are happening as expected and let the `usersync` module tests validate that they are triggered properly.

Comment (devmusings) [reply test/spec/modules/admediaBidAdapter_spec.js]: Agreed. Updated the tests.

**Comment by devmusings** [comment]
@harpere Looks like the file modules/admediaBidAdapter.js has been removed in #1936 which is causing the conflict.

**Comment by mkendall07** [comment]
@devmusings 
The adapter was removed because it doesn't comply with our new prebid 1.0 adapter requirements. You should review http://prebid.org/dev-docs/bidder-adaptor.html and submit a new new adapter that meets the requirements. I'm going to close this as it's unmergeable now.

**Pull request closed without merging**